### PR TITLE
New sleuth api for wavefront

### PR DESCRIPTION
**Pull request by marcingrzejszczak**
with this change any Sleuth compliant tracer (Brave, OTel) can work with Wavefront out of the box (e.g.https://github.com/spring-cloud-incubator/spring-cloud-sleuth-otel/pull/16)

### CODE REVIEW

**Review by jonatan-ivanov — APPROVED** [review]

File: wavefront-spring-boot/src/main/java/com/wavefront/spring/autoconfigure/WavefrontTracingSleuthConfiguration.java [hunk 1]
```diff
@@ -23,30 +24,40 @@
...
+    @Bean(BEAN_NAME)
+    @ConditionalOnMissingBean(name = BEAN_NAME)
+    @ConditionalOnBean({ MeterRegistry.class, WavefrontConfig.class, WavefrontSender.class })
+    TracingCustomizer wavefrontTracingCustomizer(WavefrontSleuthSpanHandler spanHandler) {
+      return t -> t.traceId128Bit(true).supportsJoin(false).addSpanHandler(new WavefrontSleuthBraveSpanHandler(spanHandler));
+    }
+  }
+
+
+
```
Comment (hanwavefront) [inline wavefront-spring-boot/src/main/java/com/wavefront/spring/autoconfigure/WavefrontTracingSleuthConfiguration.java:62]: nit: extra line breaks

File: wavefront-spring-boot/src/test/java/com/wavefront/spring/autoconfigure/WavefrontAutoConfigurationTests.java [hunk 2]
```diff
@@ -266,8 +266,9 @@ void tracingWithSleuthCanBeConfigured() {
         .with(sleuth())
         .run((context) -> {
           assertThat(context).hasSingleBean(TracingCustomizer.class);
-          WavefrontSleuthSpanHandler spanHandler = extractSpanHandler(context.getBean(Tracer.class));
-          Set<String> traceDerivedCustomTagKeys = (Set<String>) ReflectionTestUtils.getField(
+          WavefrontSleuthBraveSpanHandler braveSpanHandler = extractSpanHandler(context.getBean(Tracer.class));
+            WavefrontSleuthSpanHandler spanHandler = braveSpanHandler.spanHandler;
+            Set<String> traceDerivedCustomTagKeys = (Set<String>) ReflectionTestUtils.getField(
```
Comment (hanwavefront) [inline wavefront-spring-boot/src/test/java/com/wavefront/spring/autoconfigure/WavefrontAutoConfigurationTests.java:271]: nit: indentation

**Review by hanwavefront — CHANGES_REQUESTED** [review]
A couple very minor nits, LGTM otherwise.  Thanks!

**Comment by marcingrzejszczak** [comment]
Applied the changes

**Review by hanwavefront — APPROVED** [review]

**Pull request merged**